src/app/service/auth/user-auth-session.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable } from '@angular/core';
+import { OAuthService, UserInfo } from 'angular-oauth2-oidc';
+import { BehaviorSubject, from, Observable, of } from 'rxjs';
+import { catchError, switchMap } from 'rxjs/operators';
+import { AuthToken } from '../../models/auth-token';
+import {
+  IOauthSessionCheckResult,
+  oauthConfig,
+} from '../../models/oauth-session-check-result';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserAuthSessionService {
+  private _oauthUser: BehaviorSubject<UserInfo> = new BehaviorSubject<UserInfo>(
+    { sub: '' },
+  );
+  private _erroredAt: BehaviorSubject<IOauthSessionCheckResult> = new BehaviorSubject<IOauthSessionCheckResult>(
+    {
+      hasValidSession: false,
+    },
+  );
+
+  oauthSessionError$: Observable<IOauthSessionCheckResult> = this._erroredAt.asObservable();
+
+  constructor(
+    private _oauthService: OAuthService,
+    private _authService: AuthService,
+  ) {}
+
+  checkOauthSession(): Observable<IOauthSessionCheckResult> {
+    const tResult: IOauthSessionCheckResult = { hasValidSession: false };
+
+    this._oauthService.configure(oauthConfig);
+    this._oauthService.setStorage(sessionStorage);
+    this._oauthService.setupAutomaticSilentRefresh();
+
+    const tDicoveryDocObservable: Observable<boolean> = from(
+      this._oauthService.loadDiscoveryDocumentAndTryLogin(),
+    );
+
+    return tDicoveryDocObservable.pipe(
+      catchError((tErr) => {
+        tResult.error = tErr;
+        tResult.erroredAt = 'tDicoveryDocObservable';
+        return of(false);
+      }),
+      switchMap((tResp) => {
+        const tHasValidToken = this._oauthService.hasValidAccessToken();
+        if (!tResp || !tHasValidToken) {
+          this.clearSavedUserSessionData();
+          tResult.erroredAt =
+            tResp && !tHasValidToken ? 'tokenValidation' : tResult.erroredAt;
+          return of(tResult);
+        }
+
+        const tLoadUserProfileObservable = from(
+          this._oauthService.loadUserProfile(),
+        );
+        return tLoadUserProfileObservable.pipe(
+          catchError((tProfileErr) => {
+            tResult.error = tProfileErr;
+            tResult.erroredAt = 'loadUserProfile';
+            return of(false);
+          }),
+          switchMap((tProfileResp) => {
+            if (!tProfileResp) {
+              this.clearSavedUserSessionData();
+              return of(tResult);
+            }
+
+            const tToken = this._oauthService.getIdToken();
+            const tTokenObservable: Observable<AuthToken> = from(
+              this._authService.createAuthToken(tToken),
```

Comment:
you didn't update the auth service to send the Okta `id_token` in as a `okta_id_token` attribute here:
https://github.com/CruGlobal/mobile-content-admin/blob/feature/GT-934/src/app/service/auth/auth.service.ts#L30

Reply:
Spencer is going to be updating the API to take the `access_token` instead of the `id_token`, that will probably adjust the parameter name to be something else in the process

Reply:
@frett 
I did not update the `createAuthToken` method but now the app will pass the `id_token` to `createAuthToken` method and I presume this method will try to get an authentication token from the API. 
If this is successful, it will store the token to local storage and the API methods will add this token to the header of the requests. I did not change anything regarding this flow.
The only change I've made is, the app now will use `id_token` to get API authorization token and I think this is ok.
Is there anything that I'm missing here?

Reply:
the `createAuthToken` method does need a minor adjustment. when we were using the shared secret to authenticate it would send the following request:
`{"data": {"attributes": {"code":"${accessCode}"}}}`

currently when sending the `id_token` it needs to send the following request:
`{"data": {"attributes": {"okta_id_token":"${accessCode}"}}}`

But, that will probably change one more time once our API developer has a chance to update the API to take the `access_token` instead.

Reply:
This is the first mobile-content-api client that is using Okta for authentication, so there are some interoperability issues that we are sorting out as we go :)

Reply:
@frett 
Ok, I see. I missed that the parameter name would be different for getting the authorization code with okta token. 
I updated that with the latest commit. I also have a couple of more updates for displaying errors correctly.
Btw, I think it is better to use access_token for this, rather than id_token.

Reply:
yeah, it's what the API team ended up using on a different API